cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -321,6 +321,20 @@ func stringList(s string) []string {
 	return spl
 }
 
+func checkInstanceConnectivity(instanceName string, client *proxy.Client) (error) {
+	// The connectivity test right now is just a basic dial to make sure it is reachable
+	// and nothing more. This will help rule out basic network connectivity problems, though
+	// such as firewalls and the like. 
+	conn, err := client.Dial(instanceName)
+	
+	if err != nil {
+		return fmt.Errorf("Performed connectivity test to %v and got an error: %v", instanceName, err)
```

Comment:
Should say something like "Failed to connect successfully to using IP %s using instance %s: ". Ideally it should indicate if it's using public/private as well, since this would be helpful for debugging.

Reply:
Additionally, it would be helpful to find out if there is a simple way to verify we have VPC access to rule out some troubleshooting (perhaps there is a metadata server we can query?)

Reply:
1. Ack, it looks like getting the IP is a more complex since right now the proxy client abstracts all that way. I will look into what we can do there. 

2. Looks like it possible to get the private vs. public information from the Admin API. Is the right thing to add that information to the `instanceConfig` struct?

Reply:
re: getting the address; the Go errors already include this actually. Not sure if it's OK to depend on that implementation detail or not but since it's just diagnostic information maybe that's OK for the sake of simplicity.